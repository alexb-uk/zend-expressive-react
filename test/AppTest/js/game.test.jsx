import React from "react";
import renderer from "react-test-renderer";
import Game from "../../../src/App/js/components/game";

describe("<Game />", () => {
  it("renders <Game /> component", () => {
    const component = renderer.create(
      <Game />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
