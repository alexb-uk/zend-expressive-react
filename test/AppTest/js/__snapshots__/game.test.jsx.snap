// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> <Game /> history is recorded 1`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Next player: O
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Toggle Order
      </button>
    </div>
    <ol>
      <li>
        <a
          className=""
          href="#"
          onClick={[Function]}
        >
          Game start
        </a>
      </li>
      <li>
        <a
          className="selected"
          href="#"
          onClick={[Function]}
        >
          Move #1 - (1,1)
        </a>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<Game /> <Game /> history is recorded 2`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        >
          X
        </button>
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        >
          O
        </button>
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Next player: X
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Toggle Order
      </button>
    </div>
    <ol>
      <li>
        <a
          className=""
          href="#"
          onClick={[Function]}
        >
          Game start
        </a>
      </li>
      <li>
        <a
          className=""
          href="#"
          onClick={[Function]}
        >
          Move #1 - (1,1)
        </a>
      </li>
      <li>
        <a
          className="selected"
          href="#"
          onClick={[Function]}
        >
          Move #2 - (3,1)
        </a>
      </li>
    </ol>
  </div>
</div>
`;

exports[`<Game /> renders <Game /> component 1`] = `
<div
  className="game"
>
  <div
    className="game-board"
  >
    <div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
      <div
        className="board-row"
      >
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
        <button
          className="square"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="game-info"
  >
    <div>
      Next player: X
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Toggle Order
      </button>
    </div>
    <ol>
      <li>
        <a
          className="selected"
          href="#"
          onClick={[Function]}
        >
          Game start
        </a>
      </li>
    </ol>
  </div>
</div>
`;
